src/managers/config.py

```diff
@@ -97,20 +107,12 @@ def advertised_listener(self) -> str:
         return f"{self.name}://{self.host}:{self.port}"
 
 
-class ConfigManager:
-    """Manager for handling Kafka configuration."""
+class _ConfigManager:
```

Comment:
**nitpicky suggestion:** I didn't love the private class. Maybe change it to `SharedConfigManager`? `BaseConfigManager`? `CommonConfigManager`?

Reply:
**praise:** After reading through the rest of the file though, I think this choice for inheriting the base config is a really neat solution!

Reply:
Yep, on second thought, I do not think it makes that much sense in the first place to have private elements in a charm. This might be a hot take since we use private methods for event handlers (e.g. `_on_update_status`), but a charm does not expose a public API in the first place.
Anyway, sorry about the rambling, renamed in e92b42bd795ee2ae006f570f4c604bee877b437c